eugl/gqa/tasks.py

```diff
@@ -55,17 +71,455 @@
+            land = is_land_tile(self.granule, self.ocean_tile_list)
+            if land:
+                location = "{}/{}".format(self.granule, self.land_band)
+            else:
+                location = "{}/{}".format(self.granule, self.ocean_band)
+
+            h5 = h5py.File(self.input()[0].path, 'r')
+            geobox = GriddedGeoBox.from_dataset(h5[location])
+
+            landsat_scenes = intersecting_landsat_scenes(geobox_to_polygon(geobox),
+                                                         self.landsat_scenes_shapefile)
+            timestamp = acquisition_timestamp(h5, self.granule)
+            band_id = h5[location].attrs['band_id']
+            # TODO landsat sat_id
+            sat_id = 's2'
+            references = reference_imagery(landsat_scenes, timestamp, band_id, sat_id,
+                                           [self.reference_directory, self.backup_reference])
+
+            _LOG.debug("granule %s found reference images %s",
+                       self.granule, [ref.filename for ref in references])
+            vrt_file = pjoin(temp_directory, 'reference.vrt')
+            build_vrt(references, vrt_file, temp_directory)
+
+            source_band = pjoin(temp_directory, 'source.tif')
+            source_image = h5[location][:]
+            source_image[source_image == -999] = 0
+            write_img(source_image, source_band, geobox=geobox, nodata=0,
+                      options={'compression': 'deflate', 'zlevel': 1})
+
+            if land:
+                extra = ['-g', self.gverify_grid_size]
+                cmd = gverify_cmd(self, vrt_file, source_band, temp_directory, extra=extra)
+                _LOG.debug('calling gverify %s', ' '.join(cmd))
+                run_command(cmd, temp_directory)
+            else:
+                # create a set of fix-points from landsat path-row
+                points_txt = pjoin(temp_directory, 'points.txt')
+                collect_gcp(self.gverify_root_fix_qa_location, landsat_scenes, points_txt)
+
+                extra = ['-t', 'FIXED_LOCATION', '-t_file', points_txt]
+                cmd = gverify_cmd(self, vrt_file, source_band, temp_directory, extra=extra)
+                _LOG.debug('calling gverify %s', ' '.join(cmd))
+                run_command(cmd, temp_directory)
+
+            _LOG.debug('finished gverify on %s', self.granule)
+            parse_gqa(self, temp_yaml, references, band_id, sat_id, temp_directory)
+
+        except (ValueError, FileNotFoundError, CalledProcessError) as ve:
+            # failed because GQA cannot be calculated
+            _write_failure_yaml(temp_yaml, self.granule, str(ve))
+            with open(pjoin(temp_directory, 'gverify.log'), 'w') as src:
+                src.write('gverify was not executed because:\n')
+                src.write(str(ve))
+
+        self.output().makedirs()
+        shutil.copy(temp_yaml, self.output().path)
+
+        for temp_log in glob.glob(pjoin(temp_directory, '*gverify.log')):
```

Comment:
Nitpick: Can you add a comment to specify the intent of this code.
My interpretation is "If a gverify log exists move it outside of the nested work directory"

To me the for-each break after first language is imprecise and a comment will help clarify intent.

Reply:
Changed this to which probably is clearer in intent:

```python
        temp_log = glob.glob(pjoin(temp_directory, '*gverify.log'))[0]
        shutil.copy(temp_log, pjoin(self.workdir, basename(temp_log)))
```

Fails if there is no such `.log` files.